src/Server/MySQLHandler.cpp

```diff
@@ -601,12 +617,12 @@ static String showTableStatusReplacementQuery(const String & query)
     return query;
 }
 
-static String selectLimitReplacementQuery(const String & query)
+static std::optional<String> setSettingReplacementQuery(const String & query, const String & mysql_setting, const String & native_setting)
 {
-    const String prefix = "SET SQL_SELECT_LIMIT";
-    if (query.starts_with(prefix))
-        return "SET limit" + std::string(query.data() + prefix.length());
-    return query;
+    const String prefix = "SET " + mysql_setting;
+    if (0 == strncasecmp(prefix.c_str(), query.c_str(), prefix.size()))
```

Comment:
Can we please not use rather low-level C functions in C++ code (whereas `strncasecmp()` isn't even standardized)? To compare two strings case-insensitively, see e.g. [here](https://stackoverflow.com/questions/11635/case-insensitive-string-comparison-in-c).